### PR TITLE
Implement OAuth Integration for Slack App Distribution

### PR DESCRIPTION
**Pull request by TakeMeGH**
**Description**:  
This PR introduces the OAuth integration necessary for distributing the Slack app, enabling users to install and authenticate the app within their Slack workspaces.

**Changes**:
- Added OAuth authorization flow to facilitate app installation in external Slack workspaces
- Configured required scopes and permissions to support authentication
- Updated app documentation with setup instructions for new installations

**Testing**:
1. Verified OAuth handshake and successful token retrieval.
2. Tested error handling for invalid credentials and unauthorized scopes.

### CODE REVIEW

**Comment by jasonirvine76** [comment]
Is it this feature for user to connect with their slack acc? so we can see their access level?

File: main.py [hunk 1]
```diff
@@ -104,6 +118,8 @@
         reaction_event_repository,
     )
 
+    slack_view = SlackViewV1(auth_controller, config.slack_client_id, config.slack_scopes, config.admin_emails)
```
Comment (jasonirvine76) [inline main.py:121]: Refactor config to a model or typed dict, you can look at GoogleCredentials or AWSConfig

Comment (TakeMeGH) [reply main.py:121]: Sure i will take a look

**Comment by TakeMeGH** [comment]
> Is it this feature for user to connect with their slack acc? so we can see their access level?

Basically, for a Slack app to be distributed, it requires certain authentication to enable distribution. Since this will be a closed Slack app, I added a system where the app can only be added by logging in with an admin user to the CMS.

So, Broom can add the Slack app even when Broom creates another workspace.

**Comment by jasonirvine76** [comment]
> > Is it this feature for user to connect with their slack acc? so we can see their access level?
> 
> Basically, for a Slack app to be distributed, it requires certain authentication to enable distribution. Since this will be a closed Slack app, I added a system where the app can only be added by logging in with an admin user to the CMS.
> 
> So, Broom can add the Slack app even when Broom creates another workspace.

Okay, nice

**Comment by nayyara-airlangga** [comment]
Can you elaborate how the process would look like with this feature?

**Comment by TakeMeGH** [comment]
> Can you elaborate how the process would look like with this feature?

On the web, there will be an install button that redirects users to Slack, allowing them to choose which Slack workspace to install the app in. After the installation is successful, users will be redirected back to our website.

File: config/__init__.py [hunk 2]
```diff
@@ -56,6 +56,18 @@ def __init__(self) -> None:
...
+            invalid = True
+
+        self.slack_client_secret, found = self.validate_env_var("SLACK_CLIENT_SECRET")
+        if not found:
+            invalid = True
+
+        self.slack_scopes, found = self.parse_env_list("SLACK_SCOPES")
+        if not found:
+            invalid = True
+
```
Comment (nayyara-airlangga) [inline config/__init__.py:70]: I'm curious. Why do we still need this if the Slack config is added through the DTO and page of the CMS? Wouldn't this be quite redundant?

File: main.py (hunk 1 above)
Comment (nayyara-airlangga) [reply main.py:133]: Agreed on this. Often people start to use a proper struct sometimes just to define the dependencies of a service as it's easier to reason about

**Review by nayyara-airlangga — COMMENTED** [review]
Got a few questions ya

File: config/__init__.py (hunk 2 above)
Comment (TakeMeGH) [reply config/__init__.py:70]: I agree; I think this will be refactored in the next sprint.

**Review by nayyara-airlangga — APPROVED** [review]
Alright. If the plan is to refactor in the next PR as stated, then it looks good to me

**Pull request merged**